firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/PatchMutation.java

```diff
@@ -140,6 +141,19 @@ public void applyToLocalView(MutableDocument document, Timestamp localWriteTime)
     document
         .convertToFoundDocument(getPostMutationVersion(document), document.getData())
         .setHasLocalMutations();
+
+    if (squashType == MutationSquash.Type.Patch) {
```

Comment:
If we have a separate method to compute the final mutation then we likely won't need code like this or even the `MutationSquash.Type` distinction. Instead, we can just return an existing Mutation type that by itself is enough to tell the mutations apart.

Reply:
There is likely still some if/else cases, like merge a patchWithMerge to a delete, or a patch to a set?